developer-docs-site/docs/concepts/basics-gas-txn-fee.md

```diff
@@ -2,71 +2,122 @@
 title: "Gas and Transaction Fees"
 slug: "basics-gas-txn-fee"
 ---
-import BlockQuote from "@site/src/components/BlockQuote";
 
-When a transaction is executed on the Aptos blockchain, resource usage is tracked and measured in gas.
+# Gas and Transaction Fees
 
-# Introduction
+## Concept
 
-Gas ensures that all Move programs running on the Aptos blockchain eventually terminate. This bounds the computational resources that are used. Gas also provides the ability to charge a transaction fee, partly based on consumed resources during execution.
+To conduct any transaction with the Aptos blockchain on the mainnet, you are required to pay a processing fees. This includes transactions from a client application, or from a stake owner, a node operator or a voter. Think of this processing fee in the following terms: 
 
-When a client submits a transaction for execution to the Aptos blockchain, it contains a specified:
+Any blockchain, for example, the Aptos blockchain, exists in a distributed network of computing and storage resources. The Aptos mainnet is one such distributed network, holding the Aptos blockchain. 
```

Comment:
I think I mentioned this in the doc, the phrasing here is strange.

`Most blockchains including the Aptos blockchain exist in a distributed network comprised of computing and storage resources.  The Aptos mainnet is one of these networks.`